dask_glm/tests/test_unregularized.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import, division, print_function
+
+import dask.array as da
+import numpy as np
+import pytest
+
+from dask_glm.logistic import bfgs, gradient_descent, newton
+from dask_glm.utils import sigmoid, make_y
+
+
+def make_data(N, p, seed=20009):
+    '''Given the desired number of observations (N) and
+    the desired number of variables (p), creates
+    random logistic data to test on.'''
+
+    # set the seeds
+    da.random.seed(seed)
+    np.random.seed(seed)
+
+    X = np.random.random((N, p + 1))
+    col_sums = X.sum(axis=0)
+    X = X / col_sums[None, :]
+    X[:, p] = 1
+    X = da.from_array(X, chunks=(N / 5, p + 1))
+    y = make_y(X, beta=np.random.random(p + 1))
+
+    return X, y
+
+
+@pytest.mark.xfail(bfgs, reason='''
+        Early algorithm termination for unknown reason''')
```

Comment:
As is this xfails all of the tests, not just the bfgs test. 

Instead, you want to wrap xfail around the bfgs element in the list in the line below.